packages/jellyfish-api-core/__tests__/category/loan/closeVault.test.ts

```diff
@@ -43,11 +47,58 @@ describe('Loan', () => {
     ]
     oracleId = await tGroup.get(0).rpc.oracle.appointOracle(addr, priceFeeds, { weightage: 1 })
     await tGroup.get(0).generate(1)
-    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), { prices: [{ tokenAmount: '1@DFI', currency: 'USD' }] })
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), { prices: [{ tokenAmount: '1000000@DFI', currency: 'USD' }] }) // inflate DFI so that we can mint enough TSLA, we will then reset the price later
     await tGroup.get(0).rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), { prices: [{ tokenAmount: '2@TSLA', currency: 'USD' }] })
     await tGroup.get(0).rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), { prices: [{ tokenAmount: '1@DUSD', currency: 'USD' }] })
     await tGroup.get(0).generate(1)
 
+    // create loan scheme to mint loan token
+    const mintTokenLoanScheme = 'minter'
+    await tGroup.get(0).rpc.loan.setCollateralToken({
+      token: 'DFI',
+      factor: new BigNumber(1),
+      fixedIntervalPriceId: 'DFI/USD'
+    })
+
+    await tGroup.get(0).rpc.loan.setLoanToken({
+      symbol: 'TSLA',
+      fixedIntervalPriceId: 'TSLA/USD'
+    })
+
+    await tGroup.get(0).rpc.loan.createLoanScheme({
+      minColRatio: 100,
+      interestRate: new BigNumber(0.01),
+      id: mintTokenLoanScheme
+    })
+
+    await tGroup.get(0).container.generate(1)
+
+    const mintTokenVaultAddr = await loanTokenMinterGroup.generateAddress()
+    const mintTokenVaultId = await tGroup.get(0).rpc.loan.createVault({
+      ownerAddress: mintTokenVaultAddr,
+      loanSchemeId: mintTokenLoanScheme
+    })
+
+    await loanTokenMinterGroup.container.generate(20)
+
+    await loanTokenMinterGroup.rpc.loan.depositToVault({
+      vaultId: mintTokenVaultId,
+      from: loanTokenMinterAddr,
+      amount: '1@DFI'
+    })
+
+    await loanTokenMinterGroup.generate(1)
+
+    await loanTokenMinterGroup.rpc.loan.takeLoan({
```

Comment:
`loanTokenMinter` gives the idea that the loan token is being minted, which is incorrect. may we we can use `loanTokenProvider` or something like that

Reply:
naming variables hasnt been one of my strong suit and so thanks for your feedback ^^